src/Avalonia.Controls/Automation/AutomationProperties.cs

```diff
@@ -0,0 +1,539 @@
+                typeof(AutomationProperties),
+                false);
+
+        /// <summary>
+        /// Defines the AutomationProperties.IsRequiredForForm attached property.
+        /// </summary>
+        public static readonly AttachedProperty<bool> IsRequiredForFormProperty =
+            AvaloniaProperty.RegisterAttached<StyledElement, bool>(
+                "IsRequiredForForm",
+                typeof(AutomationProperties),
+                false);
+
+        /// <summary>
+        /// Defines the AutomationProperties.IsRowHeader attached property.
+        /// </summary>
+        public static readonly AttachedProperty<bool> IsRowHeaderProperty =
+            AvaloniaProperty.RegisterAttached<StyledElement, bool>(
+                "IsRowHeader",
+                typeof(AutomationProperties),
+                false);
+
+        /// <summary>
+        /// Defines the AutomationProperties.IsOffscreenBehavior attached property.
+        /// </summary>
+        public static readonly AttachedProperty<IsOffscreenBehavior> IsOffscreenBehaviorProperty =
+            AvaloniaProperty.RegisterAttached<StyledElement, IsOffscreenBehavior>(
+                "IsOffscreenBehavior",
+                typeof(AutomationProperties),
+                IsOffscreenBehavior.Default);
+
+        /// <summary>
+        /// Defines the AutomationProperties.ItemStatus attached property.
+        /// </summary>
+        public static readonly AttachedProperty<string> ItemStatusProperty =
+            AvaloniaProperty.RegisterAttached<StyledElement, string>(
+                "ItemStatus",
+                typeof(AutomationProperties));
+
+        /// <summary>
+        /// Defines the AutomationProperties.ItemType attached property.
+        /// </summary>
+        public static readonly AttachedProperty<string> ItemTypeProperty =
+            AvaloniaProperty.RegisterAttached<StyledElement, string>(
+                "ItemType",
+                typeof(AutomationProperties));
+
+        /// <summary>
+        /// Defines the AutomationProperties.LabeledBy attached property.
+        /// </summary>
+        public static readonly AttachedProperty<IControl> LabeledByProperty =
+            AvaloniaProperty.RegisterAttached<StyledElement, IControl>(
+                "LabeledBy",
+                typeof(AutomationProperties));
+
+        /// <summary>
+        /// Defines the AutomationProperties.LiveSetting attached property.
+        /// </summary>
+        public static readonly AttachedProperty<AutomationLiveSetting> LiveSettingProperty =
+            AvaloniaProperty.RegisterAttached<StyledElement, AutomationLiveSetting>(
+                "LiveSetting",
+                typeof(AutomationProperties),
+                AutomationLiveSetting.Off);
+
+        /// <summary>
+        /// Defines the AutomationProperties.Name attached attached property.
+        /// </summary>
+        public static readonly AttachedProperty<string> NameProperty =
```

Comment:
It doesn't seem any peer respects this property. If not even the base peer or control peer respects this, why include this property?

Reply:
I still need to implement quite a few properties from this class.

Reply:
See https://github.com/AvaloniaUI/Avalonia/blob/feature/ui-automation/src/Avalonia.Controls/Automation/Peers/ControlAutomationPeer.cs#L91